File: algorithms/math/isPrime.js
/**
 * @param {number} n
 * @returns {boolean}
 */
const isPrime = n => {
    if (n <= 1) {
        return false
    }

    for (let i = 2; i <= Math.sqrt(n); i++) {
        if (n % i === 0) {
            return false
        }
    }
    return true
}

export default isPrime

File: algorithms/math/is_prime.dart
import 'dart:math';

bool isPrime(int n) {
  if (n <= 1) {
    return false;
  }

  for (var i = 2; i <= sqrt(n); i++) {
    if (n % i == 0) {
      return false;
    }
  }
  return true;
}
